lib/mongo_client.js

```diff
@@ -118,6 +118,17 @@ const closeOperation = require('./operations/mongo_client_ops').closeOperation;
  * @param {number} [options.minSize] If present, the connection pool will be initialized with minSize connections, and will never dip below minSize connections
  * @param {boolean} [options.useNewUrlParser=false] Determines whether or not to use the new url parser. Enables the new, spec-compliant, url parser shipped in the core driver. This url parser fixes a number of problems with the original parser, and aims to outright replace that parser in the near future.
  * @param {boolean} [options.useUnifiedTopology] Enables the new unified topology layer
+ * @param {object} [options.autoEncryption] Optionally enable client side auto encryption
+ * @param {MongoClient} [options.autoEncryption.keyVaultClient] A `MongoClient` used to fetch keys from a key vault
+ * @param {string} [options.autoEncryption.keyVaultNamespace] The namespace where keys are stored in the key vault
+ * @param {object} [options.autoEncryption.kmsProviders] Provider details for the desired Key Management Service to use for encryption
+ * @param {object} [options.autoEncryption.kmsProviders.aws] Optional settings for the AWS KMS provider
+ * @param {string} [options.autoEncryption.kmsProviders.aws.accessKeyId] The access key used for the AWS KMS provider
+ * @param {string} [options.autoEncryption.kmsProviders.aws.secretAccessKey] The secret access key used for the AWS KMS provider
+ * @param {object} [options.autoEncryption.kmsProviders.local] Optional settings for the local KMS provider
+ * @param {string} [options.autoEncryption.kmsProviders.local.key] The master key used to encrypt/decrypt data keys
+ * @param {object} [options.autoEncryption.schemaMap] A map of namespaces to a local JSON schema for encryption
+ * @param {boolean} [options.autoEncryption.bypassAutoEncryption] Allows the user to bypass auto encryption, maintaining implicit decryption
```

Comment:
What does it mean to maintain implicit decryption if auto encryption is bypassed?

Reply:
This is the same as Dan's question about in `_command`. Imagine you want to `find` and not pay the price for encrypting a message that needs no encryption.